File: Logic Building Problems/Basic Problems/Check Even or Odd/logic2.java
// Java program to
// check for even or odd
class logic2{
    public static boolean isEven(int n){
        if ((n & 1) == 0)
            return true;
        else
            return false;
    }

    public static void main(String[] args) {
        int n = 101;
        if (isEven(n) == true)
            System.out.print("true");
        else
            System.out.print("false");
    }
}

File: Logic Building Problems/Basic Problems/Check Even or Odd/logic2.cpp
// A simple C++ program to check for even or odd logic2
#include <iostream>
using namespace std;

bool isEven(int n) {
        if ((n & 1) == 0)
            return true;
        else
            return false;
}

int main() {
    int n = 101;
     if (isEven(n) == true) 
        cout<<"true"; 
       else
        cout<<"false"; 

    return 0;
}